gitpod-io/gitpod
# [ws-daemon] Check connection to containerd socket

## Description

Mounting only the socket makes ws-daemon unavailable if containerd restarts:
- [current behavior](https://www.loom.com/share/9e750f927d8c4a11a30414c05c18bfa9)
- [with the change](https://www.loom.com/share/d1b6206c89474bbfb5e857bc36b75f30)

## Related Issue(s)
Fixes https://github.com/gitpod-io/gitpod-dedicated/issues/874

## How to test
- Integration tests should pass
- Restart `containerd` and check` ws-daemon` stays in ready
- Kill `ws-daemon` pod, and while is still starting, restart containerd. Check the pod to get ready state and not crash loop:
```
kubectl delete pods --force -l component=ws-daemon;sleep 2; \
    systemctl restart containerd;sleep 5; \
    kubectl logs -l component=ws-daemon -c ws-daemon -f
```
- Workspaces should open after the previous test

## Release Notes
```release-note
NONE
```

## Build Options:

      Experimental feature to run the build with GitHub Actions (and not in Werft).
      leeway-target=components:all
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
